access/api.go

```diff
@@ -45,9 +45,62 @@ type EventsAPI interface {
 }
 
 type ScriptsAPI interface {
-	ExecuteScriptAtLatestBlock(ctx context.Context, script []byte, arguments [][]byte) ([]byte, error)
-	ExecuteScriptAtBlockHeight(ctx context.Context, blockHeight uint64, script []byte, arguments [][]byte) ([]byte, error)
-	ExecuteScriptAtBlockID(ctx context.Context, blockID flow.Identifier, script []byte, arguments [][]byte) ([]byte, error)
+	// ExecuteScriptAtLatestBlock executes a script at latest block.
+	//
+	// CAUTION: this layer SIMPLIFIES the ERROR HANDLING convention
+	//   - All errors returned are guaranteed to be benign. The node can continue normal operations after such errors.
+	//   - To prevent delivering incorrect results to clients in case of an error, all other return values should be discarded.
+	//
+	// Expected sentinel errors providing details to clients about failed requests:
```

Comment:
nit: since there are a lot of sentinels, I suggest ordering them by likelihood. in my head, that would be
```
	//   - [access.InvalidRequestError]
	//   - [access.ResourceExhausted]
	//   - [access.DataNotFoundError]
	//   - [access.PreconditionFailedError]
	//   - [access.RequestCanceledError]
	//   - [access.RequestTimedOutError]
	//   - [access.ServiceUnavailable]
	//   - [access.InternalError]
```